test/integration/unregistered_user_authentication_test.rb

```diff
@@ -0,0 +1,72 @@
+require 'test_helper'
+
+class UnregisteredUserAuthenticationTest < ActionDispatch::IntegrationTest
+  def logged_in_user
+    User.create(username: "user", password: "password")
+    visit '/login'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.